metaconfig/src/main/scala/org/scalafmt/util/BuildTime.scala

```diff
@@ -0,0 +1,18 @@
+package org.scalafmt.util
+
+import scala.annotation.compileTimeOnly
+import scala.meta._
+
+@compileTimeOnly("Gets expanded away")
+class BuildTime extends scala.annotation.StaticAnnotation {
+
+  inline def apply(defn: Any): Any = meta {
+    defn match {
+      case defn @ Defn.Val(mods, pats, optType, rhs) if pats.size == 1 =>
```

Comment:
I originally tried to wrote it with quasiquotes, but I was tripped up by the fact that you have to explicitly type the val name as `${name: Pat.Var.Term}` (otherwise it gets typed as `Pat`, which you can't use when you come to construct the new tree).

Got it working the second time around. I agree that this is more readable.